### PR TITLE
Add scaling operation

### LINKED ISSUE #61: Data Augumentation -- scaling
This is a sub-issue for #55

This issue needs to be resolved asap.

Need to create a method for Data_Augumentation.py file for scaling the image.

### PR DESCRIPTION
**Pull request by rubyruins**
# Description:

Added a method in Data_Augumentation.py file for scaling the image based on the scaling ratio given by the user. The images are added to class/ScaledImages.

It is used in the following way:
```
flow = Flow( 'dataset_dir' , 'data_dir' )
operations = {'scale': 1.5}

flow.applyAugmentation( operations )
```

Fixes #61 

## Type of change:

- [x] New feature (non-breaking change which adds functionality)
 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

# Screenshots / Video:

Original, scale:2, scale:0.5
<p float="left">
  
  
  
</p>

### CODE REVIEW

File: flow2ml/Data_Augumentation.py [hunk 1]
```diff
@@ -146,4 +146,36 @@ def applyCrop(self,classPath):
           else:
             raise Exception("Cropping needs random parameter or list of coordinates.")
         except Exception as e:
-          print(f"Crop operation failed due to {e}")
\ No newline at end of file
+          print(f"Crop operation failed due to {e}")
+  
+  def applyScale(self,classPath):
```
Comment (chebroluharika) [inline flow2ml/Data_Augumentation.py:151]: I think you should add “random” scenario also in scaling operation

Comment (rubyruins) [reply flow2ml/Data_Augumentation.py:151]: How exactly do I include that? Scale all images by some random ratio?

**Comment by yvkrishna** [comment]
The image quality is being damaged. Can we try to restore that in anyway ?

Also if the size increases, then it would complicate the training ryt ?

**Comment by chebroluharika** [comment]
> The image quality is being damaged. Can we try to restore that in anyway ?
> 
> Also if the size increases, then it would complicate the training ryt ?

@yvkrishna scaling damages image quality, which is known(people tries to scale and superresolute images in practical scenario)
Lets take neural networks VGG16/Alexnet, it expects image to be in specific size. In that case we use this. Hence, training doesn’t get complicated.
All the thing I want to say is, we just provide functionality to scale images. After that how image needs to be modified is user’s wish. 
Hope I made some sense to you.

**Comment by yvkrishna** [comment]
> > The image quality is being damaged. Can we try to restore that in anyway ?
> > Also if the size increases, then it would complicate the training ryt ?
> 
> @yvkrishna scaling damages image quality, which is known(people tries to scale and superresolute images in practical scenario)
> Lets take neural networks VGG16/Alexnet, it expects image to be in specific size. In that case we use this. Hence, training doesn’t get complicated.
> All the thing I want to say is, we just provide functionality to scale images. After that how image needs to be modified is user’s wish.
> Hope I made some sense to you.

But while generating train, val datasets,  user provides the size to be resized ryt.

I think at that point anyway the image would get resized to the specifications provided by the vgg net or any model.

**Comment by chebroluharika** [comment]
> > > The image quality is being damaged. Can we try to restore that in anyway ?
> > > Also if the size increases, then it would complicate the training ryt ?
> > 
> > 
> > @yvkrishna scaling damages image quality, which is known(people tries to scale and superresolute images in practical scenario)
> > Lets take neural networks VGG16/Alexnet, it expects image to be in specific size. In that case we use this. Hence, training doesn’t get complicated.
> > All the thing I want to say is, we just provide functionality to scale images. After that how image needs to be modified is user’s wish.
> > Hope I made some sense to you.
> 
> But while generating train, val datasets, user provides the size to be resized ryt.
> 
> I think at that point anyway the image would get resized to the specifications provided by the vgg net or any model.

@yvkrishna we should resize image accordingly and ingest into network
Ultimately this is just implementation of rescaling which is good to go. Later steps which user should take care of

**Comment by yvkrishna** [comment]
> > > > The image quality is being damaged. Can we try to restore that in anyway ?
> > > > Also if the size increases, then it would complicate the training ryt ?
> > > 
> > > 
> > > @yvkrishna scaling damages image quality, which is known(people tries to scale and superresolute images in practical scenario)
> > > Lets take neural networks VGG16/Alexnet, it expects image to be in specific size. In that case we use this. Hence, training doesn’t get complicated.
> > > All the thing I want to say is, we just provide functionality to scale images. After that how image needs to be modified is user’s wish.
> > > Hope I made some sense to you.
> > 
> > 
> > But while generating train, val datasets, user provides the size to be resized ryt.
> > I think at that point anyway the image would get resized to the specifications provided by the vgg net or any model.
> 
> @yvkrishna we should resize image accordingly and ingest into network
> Ultimately this is just implementation of rescaling which is good to go. Later steps which user should take care of

Okay 👍

**Comment by rubyruins** [comment]
No more changes are needed then, right?

**Comment by chebroluharika** [comment]
> No more changes are needed then, right?

Yes

**Review by chebroluharika — APPROVED** [review]

**Pull request merged**